image/manifest.go

```diff
@@ -0,0 +1,108 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package image
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/snapcore/snapd/snap"
+)
+
+// The seed.manifest generated by ubuntu-image contains entries in the following
+// format:
+// <snap-name> <snap-revision>
+// The goal in a future iteration of this will be to move the generation of the
+// seed.manifest to this package, out of ubuntu-image.
+// TODO: Move generation of seed.manifest from ubuntu-image to here
+
+// ReadSeedManifest reads a seed.manifest generated by ubuntu-image, and returns
+// a map containing the snap names and their revisions.
+func ReadSeedManifest(manifestFile string) (map[string]snap.Revision, error) {
+	f, err := os.Open(manifestFile)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	revisions := make(map[string]snap.Revision)
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		tokens := strings.SplitN(line, " ", 2)
```

Comment:
should we use strings.Fields here, plus a check that the line doesn't start with whitespace? cc @mvo5

Reply:
this question is still open

Reply:
Yea left it open for @mvo to comment as well, however I agree with the suggestions

Reply:
Fixed :-)